community/kernel/src/main/java/org/neo4j/graphdb/PropertyContainer.java

```diff
@@ -126,4 +128,19 @@
      */
     // TODO: figure out concurrency semantics
     Iterable<String> getPropertyKeys();
+
+    /**
+     * Returns specified existing properties.
+     *
+     * @param keys the property keys to return
+     * @return specified properties on this property container
```

Comment:
I hope we make sure that the returned map is immutable.

What about saying that we'll throw some exception (NPE or IAE) if we get `null` keys in the array?